modules/core/server/controllers/errors.server.controller.js

```diff
@@ -7,7 +7,17 @@ var getUniqueErrorMessage = function (err) {
   var output;
 
   try {
-    var fieldName = err.errmsg.substring(err.errmsg.lastIndexOf('.$') + 2, err.errmsg.lastIndexOf('_1'));
+    var begin = 0;
+    if (err.errmsg.lastIndexOf('.$') >= 0) {
+      // support mongodb <= 3.0 (default: MMapv1 engine)
+      // "errmsg" : "E11000 duplicate key error index: mean-dev.users.$email_1 dup key: { : \"test@user.com\" }"
+      begin = err.errmsg.lastIndexOf('.$') + 2;
```

Comment:
I'm trying to understand why you're adding 2 to the index here. I just can't see it?